tests/misc_test.go

```diff
@@ -623,3 +624,10 @@ func TestTypeConversion(t *testing.T) {
 		t.Fail()
 	}
 }
+
+func TestJSONWithFixedArray(t *testing.T) {
+	var v map[string][2]float64
+	if err := json.Unmarshal([]byte(`{"a": [350, 350]}`), &v); err != nil {
+		t.Error(err)
```

Comment:
Since the bug and the fix are in `reflect` package, can we make the test lower level and use `reflect` directly? Relying on `encoding/JSON` is not optimal, because its internal implementation details may change, rendering the test ineffective.

Reply:
Done: Used reflect package directly.